src/server/dfly_main.cc

```diff
@@ -161,9 +162,28 @@ struct VersionMonitor {
   }
 
  private:
+  bool IsVersionOutdated(std::string_view remote, std::string_view current) const;
   void RunTask(SSL_CTX* ssl_ctx);
 };
 
+bool VersionMonitor::IsVersionOutdated(const std::string_view remote,
+                                       const std::string_view current) const {
+  const std::vector<absl::string_view> remote_xyz = absl::StrSplit(remote, ".");
+  const std::vector<absl::string_view> current_xyz = absl::StrSplit(current, ".");
+  CHECK_EQ(remote_xyz.size(), current_xyz.size());
+  for (size_t i = 0; i < remote_xyz.size(); ++i) {
+    size_t remote_x = 0;
+    size_t current_x = 0;
+    CHECK_EQ(absl::SimpleAtoi(remote, &remote_x), true);
+    CHECK_EQ(absl::SimpleAtoi(current, &current_x), true);
```

Comment:
You could just `CHECK()` for boolean expressions, no need to compare to `true`.
However, again, I would not crash if the `remote` part is not a legal number.